shell/components/ConfigMapSettings/Settings.vue

```diff
@@ -0,0 +1,268 @@
+      default: () => [],
+    },
+
+    values: {
+      type:     Object,
+      required: true,
+    },
+
+    labelKeyPrefix: {
+      type:    String,
+      default: 'settings'
+    },
+
+    mode: {
+      type:    String,
+      default: _EDIT
+    },
+  },
+
+  computed: {
+    settingsDisplay(): SettingDisplay[] {
+      // Init settingsDisplay with flat settings
+      const out = Object.keys(this.settings)
+        .filter((name) => !this.groups.find((g) => g.children.includes(name)))
+        .reduce((acc, name) => [
+          ...acc,
+          {
+            ...this.settings[name],
+            name,
+            options:          this.settings[name].items?.map(({ value }) => ({ value, label: this.display(`${ name }.options.${ value }`, 'label') })),
+            label:            this.display(name, 'label'),
+            description:      this.display(name, 'description'),
+            tooltipLabel:     this.settings[name].tooltip ? this.display(name, 'tooltip') : '',
+            infoLabel:        this.settings[name].info ? this.display(name, 'info') : '',
+            placeholderLabel: this.settings[name].placeholder ? this.display(name, 'placeholder') : '',
+          }
+        ], [] as SettingDisplay[]);
+
+      // Add grouped settings
+      this.groups.forEach((group) => {
+        out.push({
+          ...group,
+          label:       this.display(group.name, 'label'),
+          description: this.display(group.name, 'description'),
+          children:    group.children.reduce((acc, name) => ({
+            ...acc,
+            [name]: this.settings[name]
+          }), {})
+        });
+      });
+
+      return out.sort((a, b) => (a.weight || 0) - (b.weight || 0) || (a.label || '').localeCompare(b.label || ''));
+    },
+  },
+
+  methods: {
+    get(item: SettingDisplay) {
+      return get(this.values, item.path);
+    },
+
+    set(item: SettingDisplay, value: any) {
+      set(this.values, item.path, value);
+
+      this.update();
+    },
+
+    update() {
+      this.$emit('update:value', this.values);
+    },
+
+    display(name: string, key: 'label' | 'description' | 'tooltip' | 'info' | 'placeholder') {
+      return this.t(`${ this.labelKeyPrefix }.${ name }.${ key }`, {}, true);
+    }
+  }
+};
+</script>
+
+<template>
+  <div class="settings-container">
+    <div
+      v-for="item in settingsDisplay"
+      :key="item.name"
+      class="setting-row"
+      data-testid="cm-settings-row"
+    >
+      <div class="header mb-10">
+        <h2
+          v-if="item.label"
+          class="label"
+          :style="!item.children ? { fontSize: '18px' } : { marginBottom: '10px' }"
+        >
+          {{ item.label }}
+          <i
+            v-if="item.tooltip"
+            v-clean-tooltip="item.tooltipLabel"
+            class="icon icon-info"
+          />
+        </h2>
+        <p
+          v-if="item.description && item.type !== 'boolean'"
+          class="description text-muted mt-10"
```

Comment:
you can use the description to enrich the a11y part for each input as `aria-describedby`. I would recommend that